pkg/cloud/gcp/gcs_storage.go

```diff
@@ -42,6 +42,9 @@ const (
 	// CredentialsParam is the query parameter for the base64-encoded contents of
 	// the Google Application Credentials JSON file.
 	CredentialsParam = "CREDENTIALS"
+	// GoogleApplicationCredentialsParam is the query parameter for the file location
+	// of the Google Application Credentials JSON file. Used with implicit authentication
+	GoogleApplicationCredentialsParam = "GOOGLE_APPLICATION_CREDENTIALS"
```

Comment:
maybe I'm missing something, but do we use this anywhere? I only see us using `CredentialsParam` for specified AUTH.

Reply:
I used it to test for implicit authorization in `gcs_kms_test.go` in line 39, since from what I could tell, implicit authorization either looks for the json file in your pwd or it looks for an env variable called `GOOGLE_APPLICATION_CREDENTIALS`. However it's only something I used for testing so maybe it would be better as just a const inside `gcs_kms_test.go` rather than `gcs_storage.go`? That or maybe I'm misunderstanding how implicit authorization works.

All your other comments seem good and I will get them fixed in the next commit. Currently trying to get the roachtests to work so will let you know when they're done.

Reply:
I believe our recommendation is that if your `AUTH=implicit` then we will follow the auth protocol outlined in  https://www.cockroachlabs.com/docs/stable/use-cloud-storage-for-bulk-operations.html#google-cloud-storage which in turn links out to the google docs that recommend using the `GOOGLE_APPLICATION_CREDENTIALS` to point to a json key. The other constant in this file are actual parameters we expect the user to pass alongwith their URIs when pointing a bulk operation to gs. For example: 
`BACKUP DATABASE <database> INTO 'gs://{bucket name}/{path}?AUTH=specified&CREDENTIALS={encoded key}';`

Since we are not adding support to our URI parser for `GOOGLE_APPLICATION_CREDENTIALS` I don't think we should be adding it into this file. I also think we are getting shiny, new implicit auth docsc from @kathancox soon which will recommend users to use this env variable when using implicit auth.